include/deal.II/matrix_free/tensor_product_point_kernels.h

```diff
@@ -75,6 +75,32 @@ namespace internal
 
 
 
+  template <int dim, typename Number>
+  inline void
+  compute_values_of_array_in_pairs(
+    dealii::ndarray<Number, 2, dim>                    *shapes,
+    const std::vector<Polynomials::Polynomial<double>> &poly,
+    const Point<dim, Number>                           *p)
+  {
+    const int n_shapes = poly.size();
+
+    // Evaluate 1d polynomials and their derivatives
+    std::array<Number, 2 * dim> point, result;
+    for (unsigned int d = 0; d < dim; ++d)
+      point[d] = p[0][d];
+    for (unsigned int d = 0; d < dim; ++d)
+      point[dim + d] = p[1][d];
+    for (int i = 0; i < n_shapes; ++i)
+      {
+        poly[i].values_of_array(point, 0, &result);
+        for (unsigned int j = 0; j < 2; ++j)
+          for (unsigned int d = 0; d < dim; ++d)
+            shapes[j * n_shapes + i][0][d] = result[j * dim + d];
+      }
+  }
+
+
+
   /**
    * Specialization of above function for dim = 0. Should not be called.
```

Comment:
This comment is now confusing, because you introduced the additional function. Maybe expand to which of the above functions this one is a specialization of.